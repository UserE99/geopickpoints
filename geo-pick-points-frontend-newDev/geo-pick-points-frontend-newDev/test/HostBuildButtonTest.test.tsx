import React, { act } from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom"; // Für erweiterte Assertions wie toBeInTheDocument()
import HostBuildButton from "../src/components/HostBuildButton";

describe("HostBuildButton Component", () => {
    it("renders correctly and matches snapshot", () => {
        const mockOnClick = jest.fn(); // Mock-Funktion für onClick

        // `act` manuell um render() einhüllen
        let asFragment;
        act(() => {
            const renderResult = render(<HostBuildButton onClick={mockOnClick} />);
            asFragment = renderResult.asFragment;
        });

        // Überprüfe, ob der Button gerendert wird
        const button = screen.getByTestId("HostBuildButton");
        expect(button).toBeInTheDocument();

        // Snapshot-Test
        expect(asFragment()).toMatchSnapshot();
    });

    it("triggers onClick callback when clicked", () => {
        const mockOnClick = jest.fn(); // Mock-Funktion für onClick
        render(<HostBuildButton onClick={mockOnClick} />);

        // Simuliere einen Klick auf den Button
        const button = screen.getByTestId("HostBuildButton");
        fireEvent.click(button);

        // Überprüfe, ob die onClick-Funktion aufgerufen wurde
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });

    it("applies styles from environment variables", () => {
        // Setze Mock-Werte für Umgebungsvariablen
        process.env.REACT_APP_COLOR_PRIMARY = "#ffffff";
        process.env.REACT_APP_COLOR_SECONDARY = "#000000";

        render(<HostBuildButton onClick={() => { }} />);

        const button = screen.getByTestId("HostBuildButton");

        // Überprüfe die angewendeten Styles
        expect(button).toHaveStyle("background-color: #000000");
        expect(button).toHaveStyle("color: #ffffff");
        expect(button).toHaveStyle("border-color: #ffffff");
    });
});
