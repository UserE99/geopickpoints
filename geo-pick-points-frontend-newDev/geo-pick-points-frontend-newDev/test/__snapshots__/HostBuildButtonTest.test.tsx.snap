// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostBuildButton Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
    data-testid="HostBuildButton"
    type="button"
  >
    Create Teams
  </button>
</DocumentFragment>
`;
